zst-volatile-macro/src/lib.rs

```diff
@@ -56,34 +61,41 @@ pub fn derive_volatile(tts: TokenStream) -> TokenStream {
     .into()
 }
 
-fn parse_repr(attrs: &[Attribute]) -> Result<bool, &'static str> {
-    let repr = attrs
-        .iter()
-        .find(|a| a.path.is_ident("repr"))
-        .ok_or("VolatileStruct can only apply to #[repr(C, ...)]")?;
-
-    let repr_metas = match repr.parse_meta().unwrap() {
-        syn::Meta::List(l) => l,
-        _ => return Err("#[repr(...)] invalid format"),
-    };
-
-    // Make sure #[repr(C)] exits
-    repr_metas
-        .nested
-        .iter()
-        .find(|m| match m {
-            syn::NestedMeta::Meta(syn::Meta::Path(p)) => p.is_ident("C"),
-            _ => false,
-        })
-        .ok_or("VolatileStruct can only apply to #[repr(C, ...)]")?;
+fn parse_repr(attrs: &[Attribute]) -> syn::Result<(bool, Option<usize>, Option<usize>)> {
+    let mut repr_c = false;
+    let mut repr_align = None::<usize>;
+    let mut repr_packed = None::<usize>;
+    for a in attrs {
+        if a.path.is_ident("repr") {
+            for meta in a.parse_args_with(Punctuated::<Meta, Token![,]>::parse_terminated)? {
+                match meta {
+                    Meta::Path(path) if path.is_ident("C") => repr_c = true,
+                    Meta::Path(path) if path.is_ident("packed") => repr_packed = Some(1),
+                    Meta::List(metas) if meta.path().is_ident("packed") => {
+                        let nested: Vec<_> = metas.nested.iter().collect();
+                        match nested.as_slice() {
+                            &[NestedMeta::Lit(Lit::Int(i))] => {
+                                let n: usize = i.base10_parse()?;
+                                repr_packed = Some(n)
+                            }
+                            _ => {} // Wrong packed format
+                        }
+                    }
+                    Meta::List(metas) if meta.path().is_ident("align") => {
```

Comment:
AFAIK we can ignore `#[repr(align(n))]`. `#[repr(align(n))]` only affects the alignment of the whole struct, not the individual fields.

Reply:
Yes, it seems you're right. `PastField<B, T>::OFFSET = B::OFFSET + size_of::<T>()` actually already take `#[repr(align(n))]` on field (of child struct) into account.